bayem/distributions.py

```diff
@@ -0,0 +1,184 @@
+
+        assert len(self.mean) == len(self.precision)
+
+        if self.parameter_names is not None:
+            assert len(self.parameter_names) == len(self.mean)
+
+    def __len__(self):
+        return len(self.mean)
+
+    def index(self, parameter_name):
+        return self.parameter_names.index(parameter_name)
+
+    def named_mean(self, parameter_name):
+        return self.mean[self.index(parameter_name)]
+
+    def named_sd(self, parameter_name):
+        return self.std_diag[self.index(parameter_name)]
+
+    @property
+    def std_diag(self):
+        return np.sqrt(np.diag(self.cov))
+
+    @property
+    def covariance(self):
+        return np.linalg.inv(self.precision)
+
+    @property
+    def cov(self):
+        return self.covariance
+
+    def dist(self, dim0=0, dim1=None):
+        """
+        Exctracts a two-dimensional distribution MVN with the `dim0`th and
+        `dim1`st component of this MVN.
+        """
+        if dim1 is None:
+            return norm(loc=self.mean[dim0], scale=self.cov[dim0, dim0] ** 0.5)
+        else:
+            dim = [dim0, dim1]
+            dim_grid = np.ix_(dim, dim)
+            sub_cov = self.cov[dim_grid]
+            return multivariate_normal(mean=self.mean[dim], cov=sub_cov)
+
+    def __str__(self):
+        headers = ["name", "µ", "σ"]
+        data = [self.parameter_names, self.mean, self.std_diag]
+        data_T = list(zip(*data))
+        s = tabulate(data_T, headers=headers)
+        return s
+
+
+class Gamma:
+    def __init__(self, shape=1.0e-6, scale=1e6, name="Gamma"):
+        """
+        Creates a Gamma distribution from provided `shape` k and `scale` θ 
+        with the PDF
+
+            f(x) = 1 / [Γ(k) θ^k] x^(k-1) exp(-x/θ)
+
+        where Γ denotes the Gamma function.
+        
+        shape:
+            shape parameter k
+
+        scale:
+            scale parameter θ
+
+        name:
+            name of the distribution, e.g. to indicate "prior" or "posterior"
+
+        Notes:
+            The default values aim at creating a non-informative gamma 
+            distribution. Ideally, we would have shape=0 and scale=inf, but
+            that raises numerical issues. Other authors [Kerman, Jouni. 
+            "Neutral noninformative and informative conjugate beta and gamma 
+            prior distributions." Electronic Journal of Statistics 5 (2011): 
+            1450-1470.] propose Gamma(1/3, 0).
+        """
+        self.scale = scale
+        self.shape = shape
+        self.name = name
+
+    @property
+    def mean(self):
+        return self.scale * self.shape
+
+    @property
+    def std(self):
+        return self.scale * self.shape ** 0.5
+
+    def dist(self):
+        return gamma(a=self.shape, scale=self.scale)
+
+    def __repr__(self):
+        return f"{self.name:15} | mean:{self.mean:10.6f} | scale:{self.scale:10.6f} | shape:{self.shape:10.6f}"
+
+    @classmethod
+    def FromQuantiles(cls, x0, x1, p=(0.05, 0.95)):
```

Comment:
Just FYI:
I also implemented this in a different way [here](https://git.bam.de/mechanics/ajafari/fenics-nonlocal-damage/-/blob/master/calibrates/fwme/helper_parameters.py#L303).

Reply:
I think you took the algorithm directly from the link in the docstring, right? I only did some adjustments and performance improvements.